### PR TITLE
Feat : blog ui design with minor change in another files

### PR DESCRIPTION
**Pull request by cp-mansi-d**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import Link from "next/link";
 import ServerError from "../components/errors/serverError";
 import Avatar from "../assets/images/user.png";
 import config from "../config";
+import moment from "moment";
```
Comment (cp-priyanka-g) [inline pages/index.js:9]: Can you use another way for setting date formats, as we are not using moment library anymore.

File: pages/post/[slug].js [hunk 2]
```diff
@@ -30,11 +39,28 @@ export default function Post() {
     dispatch(fetchPostBySlug(slug));
   } else {
     post = post.attributes;
+    var publishedAt = moment(post.publishedAt).format("MMM DD YYYY");
```
Comment (cp-priyanka-g) [inline pages/post/[slug].js:42]: use simple javascript logic for date formats.

File: pages/post/[slug].js [hunk 3]
```diff
@@ -50,62 +76,94 @@ export default function Post() {
...
-                </div>
-              </div>
-              <div>
-                <div className="pb-3 text-4xl font-bold text-black">
-                  {post.title}
-                </div>
-                <div className="relative overflow-hidden transition-all bg-gray-100 rounded-md dark:bg-gray-800 aspect-video">
+            <div key={post.id} className="flex flex-col space-y-20 ">
+              <div className="grid grid-flow-row xl:grid-flow-col gap-10 xl:gap-8 w-90 h-90 rounded-3xl md:bg-[#14161E] md:py-20 md:px-10 xl:py-14 xl:px-8 ">
+                <div className="md:container w-full h-auto sm:h-[18rem] md:h-[22rem] lg:h-[30rem] xl:w-[35rem] xl:h-[20rem] 2xl:w-[42rem] 2xl:h-[23rem] ">
```
Comment (cp-priyanka-g) [inline pages/post/[slug].js:89]: can you keep all heights together in sequence like h-auto,sm:h-[18rem],md:h-[22rem]... for better readability.

File: pages/post/[slug].js [hunk 4]
```diff
@@ -30,11 +39,28 @@ export default function Post() {
...
     var authorAltText = authorData
       ? post.author_id.data.attributes.username + "images"
       : "author";
+    var tags = post.tags.data.map((tag) => {
+      var tagName = tag.attributes.name;
+      return tagName;
+    });
+    var tagsString = tags.join(", ");
+
+    // Get content_index from post.content
```
Comment (cp-priyanka-g) [inline pages/post/[slug].js:54]: remove unnecessary comments.

File: components/partials/header.js [hunk 5]
```diff
@@ -25,11 +25,11 @@ export default function Header() {
               <li className="my-2 ml-0 sm:my-0 p-0">
                 <Link
                   href="/"
-                  className={
+                  className={`relative mr-[20px] sm:mr-[40px] text-[1rem] font-normal leading-[1.125rem] tracking-[1px] md:text-[1.0625rem] md:leading-[1.5rem] lg:text-[1.125rem] lg:leading-[1.875rem] after:absolute after:top-[29px] after:md:top-[24px] after:bottom-0 after:left-0 after:w-2/4 after:h-[2px] after:bg-black-900 ${
```
Comment (cp-jagruti-a) [inline components/partials/header.js:28]: I think there should be text size of all together like text-[1rem] md:text-[1.0625rem] and like that for more readability

**Review by cp-jagruti-a — APPROVED** [review]

File: pages/post/[slug].js [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
+import { formateDate } from "../../utils";
 import md from "markdown-it";
 import Image from "next/image";
 import Avatar from "../../assets/images/user.png";
 import Script from "next/script";
 import config from "../../config";
 
 export default function Post() {
+  const [loaded, setLoaded] = useState(false);
   // get slug from URL
```
Comment (cp-priyanka-g) [inline pages/post/[slug].js:16]: remove comments from everywhere,if not require.

Comment (cp-mansi-d) [reply pages/post/[slug].js:16]: it includes the behaviour of code,I will remove it when it is no longer needed.

**Review by cp-priyanka-g — APPROVED** [review]

File: pages/index.js [hunk 7]
```diff
@@ -53,6 +54,8 @@ export default function Home() {
         ) : (
           posts.map((post, i) => {
             post = post.attributes;
+            var [date, _] = formateDate(post.publishedAt);
```
Comment (cp-sumi-k) [inline pages/index.js:57]: We are doing the same thing when we get an API response. Let's use that field only. You can replace **published_on** to **publishedAt** in `setPostFields`

File: pages/post/[slug].js [hunk 8]
```diff
@@ -30,11 +41,37 @@ export default function Post() {
...
+    var [date, _] = formateDate(post.publishedAt);
+    var publishedAt = date.replace(",", "");
     var authorData = post.author_id.data.attributes.image_url;
     var authorImage = authorData ? authorData : Avatar;
     var authorAltText = authorData
       ? post.author_id.data.attributes.username + "images"
       : "author";
+    var tags = post.tags.data.map((tag) => {
+      var tagName = tag.attributes.name;
+      return tagName;
```
Comment (cp-sumi-k) [inline pages/post/[slug].js:53]: Let's return directly without initializing variable.

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**